### PR TITLE
fixed map initialization

### PR DESCRIPTION
**Pull request by mbrueschke**

### CODE REVIEW

File: pkg/teldrvr/newrelicAPM.go [hunk 1]
```diff
@@ -81,13 +81,15 @@ type NewRelicSegmentContainer struct {
 // AddTransactionAttribute adds an attribute to the transaction
 // - Not thread safe -
 func (nrt *NewRelicTransaction) AddTransactionAttribute(key string, value any) error {
+	if nrt.attributes == nil {
+		nrt.attributes = make(map[string]any)
+	}
```
Comment (niklasgrau) [inline pkg/teldrvr/newrelicAPM.go:86]: Shouldnt this be part of some sort of constructor instead of checking it each function call?

Comment (mbrueschke) [reply pkg/teldrvr/newrelicAPM.go:86]: good input. i will rework the drivers to work like that

File: pkg/teldrvr/newrelicAPM.go [hunk 2]
```diff
@@ -36,41 +36,49 @@ func init() {
...
-type NewRelicDriver struct {
+// NewRelicAPMDriver holds all information the driver needs for telemetry
+type NewRelicAPMDriver struct {
 	NewRelicApp *newrelic.Application
 }
 
 // Start starts a transaction
-func (nrd NewRelicDriver) Start(name string) (telemetry.Transaction, error) {
-	transactionStart := nrd.NewRelicApp.StartTransaction(name)
+func (d NewRelicAPMDriver) Start(name string) (telemetry.Transaction, error) {
```
Comment (niklasgrau) [inline pkg/teldrvr/newrelicAPM.go:52]: is "d" now a generic var name for "driver"? Should we do this for all then?

Comment (mbrueschke) [reply pkg/teldrvr/newrelicAPM.go:52]: yes

File: pkg/teldrvr/local.go [hunk 3]
```diff
@@ -53,99 +60,91 @@ type LocalSegmentContainer struct {
...
+	segmentName, segmentExist := t.segmentContainer.segments[segmentID]
 	if !segmentExist {
 		return fmt.Errorf("can not add attribute to not existing segment.\nSegmentID: %s\nKey: %s\nValue: %s", segmentID, key, value)
 	}
 
-	if lt.segmentContainer.attributes == nil {
-		lt.segmentContainer.attributes = make(map[string]map[string]any)
+	if t.segmentContainer.attributes[segmentID] == nil {
+		t.segmentContainer.attributes[segmentID] = make(map[string]any)
 	}
```
Comment (niklasgrau) [inline pkg/teldrvr/local.go:98]: Idk if I am annoying.. Wouldnt it also fit to put this into the newFoobar func which you now have? Removes complexity from here again

File: pkg/teldrvr/local.go [hunk 4]
```diff
@@ -237,39 +240,39 @@ func (lt *LocalTransaction) Info(segmentID string, readCloser io.ReadCloser) err
 }
 
 // Done ends the transaction
-func (lt *LocalTransaction) Done() error {
-	log.Printf("Transaction end: %s \n", lt.transaction)
+func (t *LocalTransaction) Done() error {
```
Comment (niklasgrau) [inline pkg/teldrvr/local.go:243]: just a overall naming thing, the transaction start method is being called "Start". This one here is called "Done" and the log says "end". Is Start/End maybe a more fitting name?

File: pkg/teldrvr/local.go [hunk 5]
```diff
@@ -237,39 +240,39 @@ func (lt *LocalTransaction) Info(segmentID string, readCloser io.ReadCloser) err
...
 // SetTrace sets a trace for the transaction
-func (lt *LocalTransaction) SetTrace(trace string) error {
-	lt.trace = trace
+func (t *LocalTransaction) SetTrace(trace string) error {
+	t.trace = trace
 
 	return nil
 }
 
 // Trace returns the current ttrace for the transaction
```
Comment (niklasgrau) [inline pkg/teldrvr/local.go:266]: tttttrace. nowadays we stutter in comments?

**Review by niklasgrau — APPROVED** [review]

**Pull request merged**